integration/common/openlineage/common/provider/dbt/structured_logs.py

```diff
@@ -278,6 +299,25 @@ def parse_node_finished_event(self, event) -> RunEvent:
             outputs=outputs,
         )
 
+    def _get_attached_dataset(self, test_node_id: str) -> InputDataset:
+        """
+        gets the input the data test is attached to
+        """
+        all_nodes = {**self.compiled_manifest["nodes"], **self.compiled_manifest["sources"]}
+        test_node = all_nodes[test_node_id]
+        attached_node_id = test_node["attached_node"]
```

Comment:
Here the test is linked to its attached node whereas in the current implementation it has the arbitrary value of the last node in the parent map

https://github.com/MassyB/OpenLineage/blob/b7837beeb0154cf084cb1913c8d46688ae2eea24/integration/common/openlineage/common/provider/dbt/processor.py#L394-L397